pkg/cmd/flags/server/server.go

```diff
@@ -1,58 +1,167 @@
+					}
+					if _, err = os.Stat(addressParts[1]); err == nil {
+						err = os.Remove(addressParts[1])
+						if err != nil {
+							return nil, errfmt.Errorf("failed to cleanup gRPC listening address (%s): %v", addressParts[1], err)
+						}
+					}
+				case "tcp":
+					if len(addressParts) == 1 {
+						addressParts = append(addressParts, "4466")
+					}
+				default:
+					return nil, errfmt.Errorf("grpc protocol '%s' not supported: use tcp:<port> or unix:<socket_path>", addressParts[0])
+				}
+				server.GRPC = grpc.New(addressParts[0], addressParts[1])
+			default:
+				return nil, errfmt.Errorf("invalid server flag with value '%s': supported flags are %s, %s", flagName, HTTPAddressFlag, GRPCAddressFlag)
+			}
+		} else {
+			// Boolean flags without values
+			switch endpoint {
+			case MetricsEndpointFlag:
+				enableMetrics = true
+			case HealthzEndpointFlag:
+				enableHealthz = true
+			case PProfEndpointFlag:
+				enablePProf = true
+			case PyroscopeAgentEndpointFlag:
+				enablePyroscope = true
+			case GRPCEndpointFlag:
+				// Create default GRPC server if none exists
+				if server.GRPC == nil {
+					server.GRPC = grpc.New("unix", "/var/run/tracee.sock")
+				}
+			default:
+				return nil, errfmt.Errorf("invalid server flag '%s': supported flags are %s, %s, %s, %s, %s", endpoint, MetricsEndpointFlag, HealthzEndpointFlag, PProfEndpointFlag, PyroscopeAgentEndpointFlag, GRPCEndpointFlag)
+			}
 		}
+	}
 
-		if pprof {
-			logger.Debugw("Enabling pprof endpoint")
-			httpServer.EnablePProfEndpoint()
+	if enableMetrics || enableHealthz || enablePProf || enablePyroscope {
+		if server.HTTP == nil {
+			server.HTTP = http.New("localhost:3366")
+		}
+		if enableMetrics {
+			server.HTTP.EnableMetricsEndpoint()
 		}
-		if pyro {
-			logger.Debugw("Enabling pyroscope agent")
-			err := httpServer.EnablePyroAgent()
+		if enableHealthz {
+			server.HTTP.EnableHealthzEndpoint()
+		}
+		if enablePProf {
+			server.HTTP.EnablePProfEndpoint()
+		}
+		if enablePyroscope {
+			err = server.HTTP.EnablePyroAgent()
 			if err != nil {
-				return httpServer, err
+				return nil, err
 			}
 		}
+	}
 
-		return httpServer, nil
+	return &server, nil
+}
+
+func isValidAddr(addr string) bool {
+	// Check if the address contains a port.
+	if !strings.Contains(addr, ":") {
+		return false
+	}
+
+	// Split the address into host and port.
+	host, portStr, err := net.SplitHostPort(addr)
+	if err != nil {
+		return false
+	}
+
+	// If a host is specified, check if it's a valid IP address
+	if host != "" {
+		// Allow common hostname patterns without DNS lookup
+		if net.ParseIP(host) == nil && host != "localhost" {
+			// Basic hostname validation - just check it doesn't contain invalid characters
+			for _, char := range host {
+				if !((char >= 'a' && char <= 'z') || (char >= 'A' && char <= 'Z') ||
+					(char >= '0' && char <= '9') || char == '-' || char == '.') {
+					return false
+				}
+			}
```

Comment:
You may want to try to resolve it (LookupHost or LookupCNAME by instance) what already parses for a valid hostname, see:

https://github.com/golang/go/blob/6447ff409ac7e2a621bc8ca5c44b2eaed751fbaa/src/net/lookup.go#L469C20-L474

Reply:
Using those functions requires network IO. We should let the server handle it. Replaced custom character validation with basic sanity checks that are fast and appropriate for CLI flag validation without network I/O.